### DISCUSSION TITLE
Recently updated to 2.21.0 and am using drift_dev and MigrationStrategy. Lost Cross-platform ability for wasm.

### DISCUSSION THREAD

**Discussion by kfieldi5** [simolus3/drift#3356] [General]
Recently updated to 2.21.0 and am using drift_dev and MigrationStrategy however, since drift_dev requires sqlite3 and ffi lost my cross-platform-ness (uses web and macOS for test harness builds for backend engineers.).

Any ideas to circumvent this? I am currently trying to hack my pubspec.yaml to conditionally include those packages but have been unsuccessful thusfar.

**Comment by simolus3** [comment]
> Any ideas to circumvent this? I am currently trying to hack my pubspec.yaml to conditionally include those packages but have been unsuccessful thusfar.

The ability to compile to the web is not depending on your pubspec, but rather on the libraries (or individual Dart files) you're actually importing in your app.
In the example app, I'm using [conditional imports](https://github.com/simolus3/drift/tree/drift-2.21.0/examples/app/lib/database/connection) to guard the `drift_dev` import. That allows testing the migration on native builds while skipping that check on the web because `sqlite3` is not directly available.

On `develop`, we have new APIs that will allow the check to work on the web as well. Those haven't been released yet, but I'll hopefully get around to that soon.

**Reply by kfieldi5** [reply]
Ah I didn't originally see how to do this since it is a DB extension but this makes sense calling via VerifySelf and then making accessor methods in the conditional classes.

Thanks!

**Comment by kfieldi5** [comment]
@simolus3 dealing with these

```
$ > flutter analyze .

   info • The imported package 'drift_dev' isn't a dependency of the importing
          package • lib/services/database/native.dart:6:8 •
          depend_on_referenced_packages
   info • The imported package 'sqlite3' isn't a dependency of the importing
          package • lib/services/database/native.dart:14:8 •
          depend_on_referenced_packages
   info • The imported package 'sqlite3' isn't a dependency of the importing
          package • lib/services/database/native.dart:16:8 •
          depend_on_referenced_packages

3 issues found. (ran in 2.3s)
```

By just adding an exclusion in analysis_options.yaml -- is there a better approach?

**Reply by simolus3** [reply]
You can also just move `drift_dev` from the `dev_dependencies` to the regular `dependencies` block in your pubspec. This won't change anything about your app (like potentially including more drift code you don't need in the compiled executable).
